src/actions/clusterActions.js

```diff
@@ -417,7 +397,10 @@ export function clusterLoadDetails(clusterId, { withLoadingFlags }) {
       if (isV5Cluster) cluster.nodePools = [];
 
       const provider = getState().app.info.general.provider;
-      cluster.capabilities = computeCapabilities(cluster, provider);
+      cluster.capabilities = computeCapabilities(
+        cluster.release_version,
+        provider
+      );
 
```

Comment:
Why are we computing this in two places? Wouldn't be enough to do it just in one place?

Reply:
We are comptaiong capabilities in `clustersList`... ah, ok, this for direct access to the cluster

Reply:
Maybe we can do it in _only_ in `clusterLoadDetails`?

Reply:
I think in clusterList then only should be enough?

But maybe after upgrading a cluster, we dont call cluster list, but we do call cluster detail, so that keeps it fresh? 

In any case, I'm fairly certain we need it in the cluster listing.

Reply:
The app catalog for example doesn't need cluster details, but it does need a full list of clusters to see which can install apps on it.

So right now it does do all the cluster details calls eventually (and possibly filtered by organisation), but it doesn't wait for all the details to load in order to show the app detail page, which I think is right.

If we want to have it in clusterLoadDetails only, then we'd have to load all clusters and their details, instead of just the ones that belong to the selected organisation, when showing the app install modal.

Reply:
didn't know that... why do we need to call it on cluster listing?

Reply:
So I think if we're going to have to have it in only one place, the best place is `clustersList`, because that always happens before everything. 

The only tricky part is when creating a new cluster. Then I think we do not do a clustersList call to refresh the state, instead we do only the clusterLoadDetails for the new cluster and add the new cluster to the state manually right?

Reply:
Yes, but there is also the refreshing thing. If this is something that can be updated/modified, then I think that it makes more sense to have it in `clusterLoadDetails`.

Reply:
> why do we need to call it on cluster listing?

The install app modal shows a list of all clusters in all organisations, but it filters that list on `capabilities.canInstallApp`, because we don't want to show clusters in this list that can't install apps.

<img width="608" alt="Screenshot 2020-01-17 at 2 37 58 PM" src="https://user-images.githubusercontent.com/455309/72616358-0b12fe00-3937-11ea-9d6a-ae4909ac713c.png">

If we do it only on clusterLoadDetail, that only happens for clusters that are in the currently selected organisation.

Otherwise we'd have to load the detail for every cluster always again.

What I'm saying is, I think we can remove it from clusterLoadDetail if we are doing it now in cluster listing. But we need to check closely the case where we are creating a new cluster.

Reply:
> Yes, but there is also the refreshing thing. If this is something that can be updated/modified, then I think that it makes more sense to have it in clusterLoadDetails.

Then I think we _might_ need it in both places. 

Unless we think of a different place entirely to put this computeCapabilities. Like maybe a clusterComputeCapabilities action, which computes these values for all clusters in the current entities list, that we can then add to right place in the batched action calls.

Reply:
I'm ok with leaving this in both places then. DO you think it's useful to add a comment on why we are doing it in `clustersList`? Cause in my mind this belongs to cluster details action. WDYT?

Reply:
I'll add a comment 👍